app/gui2/src/components/GraphEditor.vue

```diff
@@ -181,6 +185,38 @@ watch(componentBrowserVisible, (visible) => {
     interactionEnded(editingNode)
   }
 })
+
+function onComponentBrowserFinished(content: String) {
+  if (content != null && graphStore.editedNodeInfo != null) {
+    graphStore.setNodeContent(graphStore.editedNodeInfo.id, content)
+  }
+  componentBrowserVisible.value = false
+  graphStore.editedNodeInfo = null
+}
+
+function getNodeContent(id: ExprId): string {
+  const node = graphStore.nodes.get(id)
+  if (node == null) return ''
+  return node.rootSpan.repr()
+}
+
+// Watch the editedNode in the graph store
+watch(
+  () => graphStore.editedNodeInfo,
+  (editedInfo) => {
+    if (editedInfo != null) {
+      const targetNode = graphStore.nodes.get(editedInfo.id)
+      const targetPos = targetNode?.position ?? Vec2.Zero()
+      const offset = new Vec2(20, 35)
+      componentBrowserPosition.value = targetPos.add(offset)
+      componentBrowserInputContent.value = getNodeContent(editedInfo.id)
+      componentBrowserVisible.value = true
+      componentBrowser.value?.focus()
```

Comment:
I believe `componentBrowser.value` will always be `undefined` here, since you have just set the `componentBrowserVisible` to true, without giving it a chance to process a render. Therefore this line does nothing. Also, `componentBrowser` ref has a wrong declared type anyway (see other comment), so it would probably error out when non-nullable.